src/spo2.c

```diff
@@ -0,0 +1,88 @@
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+#include "ls_task.h"
+#include "spo2.h"
+#include "spo2_driver.h"
+
+
+ls_task spo2_task;
+
+
+/**
+ * do_spo2() - spo2 task worker function
+ * @pvParameters:	void* to pass function parameters.
+ *
+ * This worker function retrieves an input data sample from the ADCs. This
+ * is the place where further processing of the data will happen.
+ */
+static void do_spo2(void *pvParameters);
+
+/**
+ * spo2_read() - read a single input data sample
+ * @data:	space for sample readings
+ *
+ * Read out a single data sample using the different peripherals needed.
+ */
+static void spo2_read(uint32_t *data);
+
+/**
+ * print_values() - print aquired values
+ * @data:	space for sample readings
+ *
+ * Print values to serial out that have been aquired through the peripherals
+ * for demonstration and/or debug purpuses.
+ */
+static void print_values(uint32_t *data);
```

Comment:
has nothing to do with spo2

Reply:
Other than implementing another HAL, the only use of this read function is to request values that are necessary for calculating the spo2. Its purpose is not to get all values that are potentially available to the esp core.

Reply:
The print function is for demonstration purposes. We have to think of debug switches that compile various print functions into the binary on demand. I think there should not be any print function included into any stable binary.

Reply:
esp-idf provides a pretty useful library for outputting values. 
 https://docs.espressif.com/projects/esp-idf/en/latest/api-reference/system/log.html